x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_management/components/analytics_list/action_delete.tsx

```diff
@@ -29,17 +36,96 @@ interface DeleteActionProps {
 
 export const DeleteAction: FC<DeleteActionProps> = ({ item }) => {
   const disabled = isDataFrameAnalyticsRunning(item.stats.state);
-
   const canDeleteDataFrameAnalytics: boolean = checkPermission('canDeleteDataFrameAnalytics');
 
   const [isModalVisible, setModalVisible] = useState(false);
+  const [deleteTargetIndex, setDeleteTargetIndex] = useState<boolean>(true);
+  const [deleteIndexPattern, setDeleteIndexPattern] = useState<boolean>(true);
+  const [userCanDeleteIndex, setUserCanDeleteIndex] = useState<boolean>(false);
+  const [indexPatternExists, setIndexPatternExists] = useState<boolean>(false);
+
+  const { savedObjects, notifications } = useMlKibana().services;
+  const savedObjectsClient = savedObjects.client;
+
+  const indexName = item.config.dest.index;
+
+  const checkIndexPatternExists = async () => {
+    try {
+      const response = await savedObjectsClient.find<IIndexPattern>({
+        type: 'index-pattern',
+        perPage: 10,
+        search: `"${indexName}"`,
+        searchFields: ['title'],
+        fields: ['title'],
+      });
+      const ip = response.savedObjects.find(
+        (obj) => obj.attributes.title.toLowerCase() === indexName.toLowerCase()
+      );
+      if (ip !== undefined) {
+        setIndexPatternExists(true);
+      }
+    } catch (error) {
+      const { toasts } = notifications;
+
+      toasts.addDanger(
+        i18n.translate(
+          'xpack.ml.dataframe.analyticsList.errorWithCheckingIfIndexPatternExistsNotificationErrorMessage',
+          {
+            defaultMessage:
+              'An error occurred checking if index pattern ${indexPattern} exists: {error}',
+            values: { indexPattern: indexName, error: JSON.stringify(error) },
+          }
+        )
+      );
+    }
+  };
+  const checkUserIndexPermission = () => {
+    try {
+      const userCanDelete = canDeleteIndex(indexName);
+      if (userCanDelete) {
+        setUserCanDeleteIndex(true);
+      }
+    } catch (error) {
+      const { toasts } = notifications;
+      toasts.addDanger(
+        i18n.translate(
+          'xpack.ml.dataframe.analyticsList.errorWithCheckingIfUserCanDeleteIndexNotificationErrorMessage',
+          {
+            defaultMessage:
+              'An error occurred checking if user can delete ${destinationIndex} exists: {error}',
```

Comment:
I don't think this message needs `exists` at the end. Should just be `An error occurred checking if user can delete ${destinationIndex} ` I think.

Reply:
Done in https://github.com/elastic/kibana/pull/66934/commits/5bd97e9c466ad1a88d32da4f432583d5ddc3aed0